transformations.go

```diff
@@ -435,3 +436,130 @@ func WriteCodonJSON(codontable CodonTable, path string) {
 	file, _ := json.MarshalIndent(codontable, "", " ")
 	_ = ioutil.WriteFile(path, file, 0644)
 }
+
+/******************************************************************************
+Dec, 17, 2020
+
+Compromise + Add codon table stuff begins here
+
+
+== Compromise tables ==
+Basically, I want to codon optimize a protein for two or more organisms.
+In order to do that, I need to be able to generate a codon table that
+is a compromise between the codon tables between two different organisms.
+
+The method is fairly simple: standardize codon counts so the weights are
+equal between both organisms, then add them together. In addition, have
+a variable percentage for removing rare codons (this makes compromise
+tables lossy).
+
+Simple code, but very powerful if it can be used to encode genes for
+multiple organisms.
+
+== Add tables ==
+Some organisms have multiple chromosomes. We need to add em all up
+to get an accurate codon table (different than compromise tables,
+since these are all already balanced).
+
+Godspeed,
+
+Keoni
+******************************************************************************/
+
+// CompromiseCodonTable takes 2 CodonTables and makes a new CodonTable
+// that is an equal compromise between the two tables.
+func CompromiseCodonTable(firstCodonTable CodonTable, secondCodonTable CodonTable, cutOff float64) (CodonTable, error) {
```

Comment:
I understand the purpose of this function but I don't really understand how it works. Could you clean up some names and add comments?

Reply:
Added more comments. I think the names are pretty good in this one, but please tell me if any are unclear. I think the comments might clear things up. (changed a few things so it would be easier to reason about in latest commit, tests still pass and everything and the code acts the same)